src/Speckle_filter/median_filter.jl

```diff
@@ -0,0 +1,22 @@
+@doc """
+    median filter
+"""
+function speckle_median_filter(image::Matrix{T} where T<: Real ,filter_size::Vector{N} where N<:Integer =[3,3])::Matrix{Real}
+    @assert length(filter_size) ==2  "Filter size must be [int,int] for height and widht of filter."
+    @assert iseven(filter_size[1]) !=true  "Filter height must be uneven, is $(filter_size[1])."
+    @assert iseven(filter_size[2]) !=true  "Filter width must be uneven, is $(filter_size[2])."
+    @assert size(image)[1]>=filter_size[1]  "Image must be larger than filter. Image height $(size(image)[1]) filter height $(filter_size[1])"
+    @assert size(image)[2]>=filter_size[2]  "Image must be larger than filter. Image width $(size(image)[2]) filter width $(filter_size[2])"
```

Comment:
consider reducing assserts as explained earlier. You could also make a helper function to do this, it is usually better practice than copying the same lines of code over at over. But I might be wrong on this, I am not the pro software dev :D